api/src/pcapi/alembic/versions/20240502T140549_ca19ecfe0c74_create_postgres_users.py

```diff
@@ -0,0 +1,94 @@
+"""create users for the tools"""
+
+import secrets
+
+from alembic import op
+
+from pcapi import settings
+from pcapi.utils import secrets as secrets_utils
+
+
+# pre/post deployment: post
+# revision identifiers, used by Alembic.
+revision = "ca19ecfe0c74"
+down_revision = "34a1a2acff4f"
+branch_labels: tuple[str] | None = None
+depends_on: list[str] | None = None
+
+
+def upgrade() -> None:
+    db_name = settings.DATABASE_URL.split("/")[-1]  # type: ignore [union-attr]
+    # create and configure user for backend
+    op.execute(
+        f"CREATE USER backend WITH LOGIN NOINHERIT PASSWORD '{secrets_utils.get('PASSWORD_POSTGRESQL_BACKEND', secrets.token_hex(20))}';"
+    )
+    op.execute("ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT, INSERT, UPDATE, DELETE ON TABLES TO backend;")
+    op.execute("GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA public TO backend;")
```

Comment:
Role RW. On aura aussi besoin d'un role pour teleport, donc autant l'intégré tout de suite d'autant que ça reviens déjà 3 fois ici, ex https://dev.to/sandipbasnet/create-user-with-read-only-permission-in-postgresql-1g4h

Reply:
peut être aussi tester la création d'une table après les requêtes, pour tester ça: 

**grant access to future tables:**

> ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT ON TABLES TO readaccess;